projects/libs/core-library-typescript/src/block-storage/block-storage.ts

```diff
@@ -82,8 +82,11 @@ export class BlockStorage {
   public async getBlocks(fromLastBlockHeight: number): Promise<types.Block[]> {
     const blocks: types.Block[] = [];
 
-    for (let i = fromLastBlockHeight; i < this.lastBlock.header.height; ++i) {
-      blocks.push(await this.getBlock(i + 1));
+    // lastBlock is undefined when the service did not initialize the logic/load() did not run yet (race condition most likely)
```

Comment:
It's not a very good idea to return empty blocks if we actually supposed to have the blocks, I think maybe we should either reject the promise or run `.load()` and wait until it's done.